### DISCUSSION TITLE
meta win + 1 2 3 number keys

### DISCUSSION THREAD

**Discussion by markifi** [lxqt/lxqt#2467] [Q&A]
i love lxqt but there's just one behaviour i can't get used to and i tried for months now. in kde plasma and microsoft windows you pin the programs (not necessary) and can use the meta/windows plus number key combination to consistently land back on the same program e.g., i have the terminal on windows+1, browser on 2, file manager on 3 and media player on 4 on all my computers. this however in lxqt activates the windows on the taskbar and their order is dynamic so if i close one the keys to get back the same place are no longer the same. can you give me some advice on how to configure it so these keys always activate the same functions? ctrl+alt+t, ctrl+alt+i, ctrl+alt+e are useful but not quite the same as they open new windows instead of activating the instance already open

**Comment by stefonarch** [comment]
Do you tie the programs to workspaces?
I do, so I've Mail, Net,terminal, social  ecc and can switch in many ways between them, including activating workspace by shortcut.

This is more a window manager setting than an LXQt one, actually I don't see with which WM you could pin a shortcut to an application window. 

Did some test and ~~I see it's actually impossible to assing meta+0 instead of F12 to qterminal dropdown~~ inside LXQt shortcut keys change it to some other combination, just disabling doesn't make it available. 

If your using openbox something like this in `rc.xml` could work:

```
 <keybind key="W-0">
    <action name="ForEach">
      <query target="default">
         <class>qterminal</class>
       </query>
    <then>
     <action name="Focus"/>
     <action name="Raise"/>
     </then>
    </action>
  </keybind>


<keybind key="W-2">
 <action name="ForEach">
  <query target="default">
    <title>*Firefox*</title>
  </query>
  <then>
     <action>Focus</action>
     <action>Raise</action>
 </action>
</keybind>

```

**Reply by stefonarch** [reply]
Remembered `wmctrl`:

```
ACTIONS
       The  following  command  line arguments can be specified to invoke a wmctrl action. Only one action can be executed with the invocation of the wmctrl
       command.

       -a <WIN>
              Switch to the desktop containing the window <WIN>, raise the window, and give it focus.


```
So shortcuts for `wmctrl -a firefox` and similar should work.

EDIT: did some tests, it looks like more complicated as the` -a` option takes the title and `-x` didn't work.  On wayland (labwc) it's as easy as `wlrctl toplevel focus firefox
`
